src/transactions/types.ts

```diff
@@ -77,9 +77,9 @@ export type AnyRawTransactionInstance = RawTransaction | MultiAgentRawTransactio
  * Optional options to set when generating a transaction
  */
 export type InputGenerateTransactionOptions = {
-  maxGasAmount?: AnyNumber;
-  gasUnitPrice?: AnyNumber;
-  expireTimestamp?: AnyNumber;
+  maxGasAmount?: number;
+  gasUnitPrice?: number;
+  expireTimestamp?: number;
```

Comment:
I guess the only issue is setting "no expiry", as well as anytime after January 19th 2038, it may fail if it's not represented as a bigint or a string...

Would a string work?

Reply:
JavaScript uses a 64-bit floating-point number to represent dates. Jan 19, 2038 shouldn't be affected with use of number. 

Yes a string _would_ work, but would have to be converted to number elsewhere. If your only issue was the 32-bit overflow, would be good to keep as number.

Reply:
Okay, let's make it a number, I'm convinced since the max safe integer in microseconds is 2255.

Can we document that it must be a `SafeInteger`?  Because I guess the bounds are still the same problem anyways, where the number could be put in here negative, and that would also fail.

For a future update let's add a function that we can use to verify `isUnsignedInt` which would verify the bounds `0-> Number.MAX_SAFE_NUMBER`.  Then we won't get weird float values, and we can add it more places to provide better help with it.